lib/shoryuken/queue.rb

```diff
@@ -5,7 +5,11 @@ class Queue
     def initialize(client, name)
       self.name   = name
       self.client = client
-      self.url    = client.get_queue_url(queue_name: name).queue_url
+      begin
+        self.url = client.get_queue_url(queue_name: name).queue_url
+      rescue Aws::SQS::Errors::NonExistentQueue
+        raise StandardError.new "Non existing queue: #{name}. Please create it in advance in SQS."
```

Comment:
I would keep it transparent to aws-sdk as possible. I agree `NonExistentQueue` is a generic issue from aws-sdk, it can be that the queue does not exist or that you don't have permissions to it.

> The default error makes no mention to the queue which is failing

Maybe we could `logger.error(...)` and `raise e` wdyt?